### PR TITLE
Update XUnit and Microsoft.DotNet.XUnitAssert to 2.6.6

### PR DESCRIPTION
**Pull request by akoeplinger**
Needed to essentially revert https://github.com/xunit/assert.xunit/commit/455865ac846c0812e80ffb1c4a46b9d5d35ff828 to make it AOT compatible.

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/XUnit/XUnit.targets [hunk 1]
```diff
@@ -13,11 +13,6 @@
     <PackageReference Include="xunit.runner.visualstudio" Version="$(XUnitRunnerVisualStudioVersion)" IsImplicitlyDefined="true" PrivateAssets="all" Publish="true"/>
   </ItemGroup>
 
-  <!-- Workaround for https://github.com/xunit/xunit/issues/1651 -->
-  <ItemGroup>
-    <None Remove="$(NuGetPackageRoot)xunit.runner.visualstudio\$(XUnitRunnerVisualStudioVersion)\build\net20\..\_common\xunit.abstractions.dll" />
-  </ItemGroup>
-
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/XUnit/XUnit.targets:20]: Interesting, why did you remove this? Is this no longer the case?

File: src/Microsoft.DotNet.XUnitConsoleRunner/src/ConsoleRunner.cs [hunk 2]
```diff
@@ -178,7 +178,7 @@ void PrintHeader()
         {
             var platform = System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription;
 
-            Console.WriteLine($"Microsoft.DotNet.XUnitConsoleRunner v2.5.0 ({IntPtr.Size * 8}-bit {platform})");
+            Console.WriteLine($"Microsoft.DotNet.XUnitConsoleRunner v{typeof(ConsoleRunner).Assembly.GetName().Version} ({IntPtr.Size * 8}-bit {platform})");
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.XUnitConsoleRunner/src/ConsoleRunner.cs:181]: Interesting. For GenAPI we use the following API: https://github.com/dotnet/sdk/blob/bfd2919bc446cad68d11de90cec4025d3683591c/src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI.Tool/Program.cs#L83

Which one is better?

File: src/Microsoft.DotNet.XUnitConsoleRunner/src/ConsoleRunner.cs [hunk 3]
```diff
@@ -388,13 +388,15 @@ void PrintUsage(IReadOnlyList<IRunnerReporter> reporters)
 
                         reporterMessageHandler.OnMessage(new TestAssemblyExecutionStarting(assembly, executionOptions));
 
+#pragma warning disable CS0618 // Delegating*Sink types are marked obsolete, but we can't move to ExecutionSink yet
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.XUnitConsoleRunner/src/ConsoleRunner.cs:391]: Please file an issue and add a link to it here

File: src/Microsoft.DotNet.Arcade.Sdk/tools/XUnit/XUnit.targets (hunk 1 above)
Comment (akoeplinger) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/XUnit/XUnit.targets:20]: yeah version xunit.runner.visualstudio 2.4.2 removed the `net20` folder and we bumped to that with https://github.com/dotnet/arcade/commit/d8f74bfbcf452ecb64e505d04e874b51725798f5

Comment (akoeplinger) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/XUnit/XUnit.targets:20]: I mean it's possible that we still hit the same problem but this code has been a no-op for the last four years

File: src/Microsoft.DotNet.XUnitConsoleRunner/src/ConsoleRunner.cs (hunk 2 above)
Comment (akoeplinger) [reply src/Microsoft.DotNet.XUnitConsoleRunner/src/ConsoleRunner.cs:181]: `Assembly.Location` doesn't work on NativeAOT since there's no .dll file on disk

File: src/Microsoft.DotNet.Arcade.Sdk/tools/XUnit/XUnit.targets (hunk 1 above)
Comment (ViktorHofer) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/XUnit/XUnit.targets:20]: I just verified, this is still the case. Instead of removing, can you please change the None line to:

`<None Remove="$(NuGetPackageRoot)xunit.runner.visualstudio\$(XUnitRunnerVisualStudioVersion)\build\net462\xunit.abstractions.dll" />`

Comment (ViktorHofer) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/XUnit/XUnit.targets:20]: I'm not sure why we never got this fixed and decided to workaround it, but I just left a comment in the issue to understand some of the design choices before attempting to fix it.

**Review by ViktorHofer — APPROVED** [review]
